src/clj/ataru/hakija/hakija_application_service.clj

```diff
@@ -77,28 +77,68 @@
     (and (= state "processing")
          (:is-jatkuva-haku? (:tarjonta tarjonta-info)))))
 
+(defn- get-hakuaika-end
+  [application tarjonta-service]
+  (when tarjonta-service
+    (:end (get-hakuaikas tarjonta-service application))))
+
+(defn- only-attachments-editable?
+  [answer application tarjonta-service]
+  (let [hakuaika-end (get-hakuaika-end application tarjonta-service)]
+    (and (when hakuaika-end
+           (after-apply-end-within-10-days? hakuaika-end))
+         (not= (:fieldType answer) "attachment"))))
+
+(defn- dummy-answer-to-unanswered-question
+  [{:keys [id fieldType label]}]
+  {:key       id
+   :fieldType fieldType
+   :label     label
+   :value     ""})
+
+(defn- filter-questions-without-answers
+  [answers-by-key form-fields]
+  (filter (fn [answer]
+            (and (not (some #{(keyword (:id answer))} (keys answers-by-key)))
+                 (some #{(:fieldType answer)} ["textField"
```

Comment:
I'm a bit worried about these kinds of "magic lists" that can easily be left forgotten when e.g. creating new types of fields in the future. Could this either check for `:fieldClass` with value `"formField"` or if not, be extracted somehow?

Reply:
I was thinking about the same, i'll extract it..